controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -69,19 +84,37 @@ class ControllerManager : public rclcpp::Node
     typename T,
     typename std::enable_if<std::is_convertible<
       T *, controller_interface::ControllerInterface *>::value, T>::type * = nullptr>
-  std::shared_ptr<controller_interface::ControllerInterface>
-  add_controller(std::shared_ptr<T> controller, std::string controller_name)
+  controller_interface::ControllerInterfaceSharedPtr
+  add_controller(
+    std::shared_ptr<T> controller, std::string controller_name,
```

Comment:
Actually I believe this is one of these instances where it's faster to not pass by const reference. Since you'll make  a copy anyway later, with copy ellision it boils down to a single copy. With reference it's an address copy + a copy of the string.

All this doesn't matter anyway, because in this case the method is inlined, so the compiler will do whatever it thinks is best.

I can still change it for uniformity if you want.

Reply:
I'd prefer const ref